java/memory/memory-core/src/main/java/org/apache/arrow/memory/DefaultAllocationManagerOption.java

```diff
@@ -130,4 +142,18 @@ private static AllocationManager.Factory getNettyFactory() {
           " No DefaultAllocationManager found to instantiate an NettyAllocationManager", e);
     }
   }
+
+  private static AllocationManager.Factory getFfmFactory() {
+    try {
+      return getFactory("org.apache.arrow.memory.FfmAllocationManager");
+    } catch (RuntimeException e) {
+      if (majorVersion < 21) {
```

Comment:
Would we also have to detect that preview features are actually turned on and the underlying FFM libraries are available in that specific Java 21+ VM instance?

Reply:
Yes, good catch. More helpful checks would be nice.